Engine/Results/BaseResultsHandler.cs

```diff
@@ -156,6 +157,17 @@ public abstract class BaseResultsHandler
         /// <remarks>Update frequency of notification packets</remarks>
         protected TimeSpan NotificationPeriod { get; set; }
 
+        /// <summary>
+        /// Directory location to store results
+        /// </summary>
+        protected string ResultsDestinationFolder;
+
+        /// <summary>
+        /// Root filename to store results
+        /// </summary>
+        /// <remarks>Should not include the file extension here</remarks>
+        protected string ResultsDestinationFilePrefix => Config.Get("results-destination-file-prefix", AlgorithmId);
```

Comment:
Wondering if the `AlgorithmId` could have the same effect as `ResultsDestinationFilePrefix`? Maybe we need a way of setting the `AlgorithmId` (BacktestId and DeployId at JobQueue.cs) to be different than `AlgorithmTypeName`, but maybe defaulting to it 🤔. That way we would get uniqueness across all users of `AlgorithmId`

Reply:
@Martin-Molinero - I'm not quite sure what to do with this one? Can you give me an example of what you're thinking? (Happy to make the change)

Reply:
I was thinking something like the following https://github.com/QuantConnect/Lean/commit/b49fb478c98ad3e83fee728b75bd266711573033, unit tests need an update too I believe